glsk/glsk-document-cse/src/main/java/com/powsybl/glsk/cse/CseGlskShiftKey.java

```diff
@@ -99,8 +99,15 @@ private void initCommonMemberVariables(BlockWrapper blockWrapper, BusinessTypeLi
             blockWrapper.getMaximumShift().map(BigDecimal::doubleValue)
                 .ifPresent(ms -> this.maximumShift = ms);
         } else {
-            blockWrapper.getFactor().map(BigDecimal::doubleValue)
-                .ifPresent(q -> this.quantity = q);
+            blockWrapper.getFactor()
```

Comment:
From my point of view, putting the comparison of `sumBlockFactors` with 0 and 1 in a final variable allows us to write some clearer code. You may not share the same opinion :smiley:.

A proposal below:
```java
            final boolean sumIsZeroOrOne =
                BigDecimal.ZERO.compareTo(sumBlockFactors) == 0
                    || BigDecimal.ONE.compareTo(sumBlockFactors) == 0;

            blockWrapper.getFactor()
                .map(factor -> sumIsZeroOrOne ? factor : factor.divide(sumBlockFactors))
                .ifPresent(q -> this.quantity = q.doubleValue());
```